pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,171 @@
+their reliance on simple lexical comparisons, and thus this PEP has
+been born.
+
+
+Rationale
+=========
+
+Many Python packages are written with portability in mind.
+
+For many packages this means they aim to support a wide range of
+Python releases. If they depend on libraries such as ``argparse`` -
+which started as external libraries, but later got incorporated into
+core - specifying a single set of requirements is difficult, as the
+set of required packages differs depending on the version of Python in
+use.
+
+For other packages, designing for portability means supporting
+multiple operating systems. However, the significant differences
+between them may mean that particular dependencies are only needed on
+particular platforms (relying on ``pywin32`` only on Windows, for
+example)"
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to list requirements that are
+specific to a particular environment.
+
+Examples
+========
+
+Here are some examples of such markers inside a requirements.txt::
+
+   pywin32 >=1.0 ; sys_platform == 'win32'
+   unittest2 >=2.0,<3.0 ; python_version == '2.4' or python_version == '2.5'
+   backports.ssl_match_hostname >= 3.4 ; python_version < '2.7.9' or (python_version >= '3.0' and python_version < '3.4')
+   hawkey ; 'linux' in sys_platform
+
+And here's an example of some conditional metadata included in
+setup.py for a distribution that requires PyWin32 both at runtime and
+buildtime when using Windows::
+
+   setup(
+     install_requires=["pywin32 > 1.0 : sys.platform == 'win32'"],
+     setup_requires=["pywin32 > 1.0 : sys.platform == 'win32'"]
+     )
+
+
+Micro-language
+==============
+
+The micro-language behind this is as follows. It compares::
+
+* strings with the ``==`` and ``in`` operators (and their opposites)
+* version numbers with the ``<``, ``<=``, ``>=``, and ``<`` operators
+  in addition to those supported for strings
+
+The usual boolean operators ``and`` and ``or`` can be used to combine
+expressions, and parentheses are supported for grouping.
+
+The pseudo-grammar is ::
+
+    MARKER: EXPR [(and|or) EXPR]*
+    EXPR: ("(" MARKER ")") | (STREXPR|VEREXPR)
+    STREXPR: STRING [STRCMPOP STREXPR]
+    STRCMPOP: ==|!=|in|not in
+    VEREXPR: VERSION [VERCMPOP VEREXPR]
+    VERCMPOP: (==|!=|<|>|<=|>=)
+
+
+``SUBEXPR`` is either a Python string (such as ``'win32'``) or one of
+the ``Strings`` marker variables listed below.
+
+``VEREXPR`` is a PEP-0440[3] version identifier, or one of the
+``Version number`` marker variables listed below. Comparisons between
+version numbers are done using PEP-0440 semantics.
+
+
+Strings
+-------
+
+* ``os_name``: ``os.name``
+* ``sys_platform``: ``sys.platform``
+* ``platform_release``: ``platform.release()``
+* ``implementation_name``: ``sys.implementation.name``
+* ``platform_machine``: ``platform.machine()``
+* ``platform_python_implementation``: ``platform.python_implementation()``
```

Comment:
It is probably useful to link to the docs for the platform module to explain what these values return, or maybe to give a sample value for each, or perhaps just give a simple description. Otherwise it's not clear what we might expect in  "platform_machine" vs  "platform_release". It's trivial to find out but it'd be nice if people could tell just by reading the PEP without needing to go and run bits of code